core/genesis.go

```diff
@@ -477,22 +491,23 @@ func (g *Genesis) IsVerkle() bool {
 
 // ToBlock returns the genesis block according to genesis specification.
 func (g *Genesis) ToBlock() *types.Block {
-	var root common.Hash
+	var stateRoot, storageRootMessagePasser common.Hash
 	var err error
 	if g.StateHash != nil {
 		if len(g.Alloc) > 0 {
 			panic(fmt.Errorf("cannot both have genesis hash %s "+
 				"and non-empty state-allocation", *g.StateHash))
 		}
-		root = *g.StateHash
-	} else if root, err = hashAlloc(&g.Alloc, g.IsVerkle()); err != nil {
+		// TODO - need to get the storage root of the L2ToL1MessagePasser contract?
+		stateRoot = *g.StateHash
```

Comment:
Going to need some input here @protolambda . If we find just the [StateHash](https://github.com/ethereum-optimism/superchain-registry/blob/main/superchain/superchain.go#L497-L500), is there a way to get the storage root of the message passer?

Reply:
The `stateHash` is only there for pre-bedrock chains, which also means pre-isthmus. So no need to retrieve/set the withdrawals hash in those chains. You can update the comment to reflect that dependency. And maybe add a error or panic if someone tries to use state-hash for a genesis that is post-isthmus.